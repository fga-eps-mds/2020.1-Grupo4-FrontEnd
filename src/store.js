import { configureStore } from '@reduxjs/toolkit';
import usersReducer from './slices/usersSlice';
import tutorialSlice from './slices/tutorialSlice'

export default configureStore({
  reducer: {
    users: usersReducer,
    tutorial: tutorialSlice
  }
});