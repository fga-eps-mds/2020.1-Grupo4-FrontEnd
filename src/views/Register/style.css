.register-form {
  
    padding: 5%;
    max-height: 900px;
    font-family: Overpass;
    background-color: white;
    border-radius: 10px;
    width: 698px;
    height: 717px;
    margin: auto;
    margin-top: 10vw;
    margin-bottom: 10vw;
}

.register-form h1 {
    font-family: Overpass;
    margin-left: 80px;
    font-size: 48px;
}
.register-form label form input[type="text"], input[type="password"]{
    outline: none;
    width: 200px;
    display: block;
    margin-bottom: 20px;
    border-width: 0.2px;
    box-sizing: border-box;
    padding: 5px;
    border-radius: 10px;
}

.register-inputs {
    max-width: 600px;
    margin-top: 40px;
    display: grid;
    grid-template-columns: 2fr 2fr;
}

.register-inputs .register-grid1{
    margin-left: 80px;
}
.register-inputs .register-grid2{
    margin-left: 20px;
}

.register-options {
    display: flex;
    flex-direction: column;
    margin-left: 80px;
}
.register-options p {
    font-size: 16px;
    color: var(--title-color);
    font-family: Overpass;
}
.register-options label {
    margin-right: 10px;
}
.register-options label input[type="radio"], input[type="checkbox"]{
    margin: 0;
}
.register-options label{
    display: inline-block;
    margin-bottom: 30px;
}

.register-form .button {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 38px;
    width: 200px;
    margin-left: 216.5px;
}

@media(max-width: 800px) {
    .register-form {
        width: 90vw;
        height: 46rem;
    }
    .register-inputs {
       display: flex;
       flex-direction: column;
       align-items: center;
       margin: 0;
    }
    .register-inputs .register-grid1{
        margin-left: 4vw;
    }
    .register-inputs .register-grid2{
        margin-left: 4vw;
    }
    .register-form h1 {
        margin: 0;
       text-align: center;
    }

    .register-options {
        display: inline;
        margin-left: 0px;
    }
    .user-type {
        text-align: center;
    }
    .register-options p{
        margin: 0;
    }
    .register-options label{
        margin-left: 0;
        display: inline-block;
        margin-bottom: 0.2rem;
    }
    .register-form .button {
        margin: 0 auto;
        margin-top: 4vh;
        height: 30px;
    }
}
