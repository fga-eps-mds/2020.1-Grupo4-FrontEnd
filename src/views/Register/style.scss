#register {
    padding: 5%;
    background: linear-gradient(309.23deg, rgba(136, 56, 255, 0.95) 3.17%, rgba(233, 66, 255, 0.5) 114.71%);
}

.register__form {
    padding: 5%;
    max-height: 900px;
    font-family: Overpass;
    background-color: white;
    border-radius: 10px;
    width: 698px;
    height: 717px;
    margin: auto;
    margin-top: 10vw;
    margin-bottom: 10vw;

    h1 {
        font-family: Overpass;
        margin-left: 80px;
        font-size: 48px;
    }

    label form input[type="text"], input[type="password"] {
        outline: none;
        width: 200px;
        display: block;
        border-width: 0.2px;
        box-sizing: border-box;
        padding: 5px;
        border-radius: 10px;
        padding: 4px 4px;
        }
      
}



.register__inputs {
    max-width: 600px;
    margin-top: 40px;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 30px;
    p {
        margin-bottom: 0;
    }
    padding-right: 80px;
    padding-left: 80px;
    
}
  

.register__options {
    display: flex;
    flex-direction: column;
    margin-left: 80px;

    p {
        font-size: 16px;
        color: var(--title-color);
        font-family: Overpass;
        margin-top: 20px;
    }

    label {
        margin-right: 10px;
        display: inline-block;
        margin-bottom: 30px;
    }

    label input[type="radio"], input[type="checkbox"]{
        margin: 0;
    }

}

.register__form .button {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 38px;
    width: 200px;
    margin-left: 204px;
}

@media(max-width: 800px) {
    .register__form {
        width: 90vw;
        height: 46rem;
    }
    .register__inputs {
       display: flex;
       flex-direction: column;
       align-items: center;
       margin: 0;
       padding-left: 0;
       padding-right: 0;
    }
  
    .register__form h1 {
        margin: 0;
        text-align: center;
    }

    .register__options {
        display: inline;
        margin-left: 0px;

        p {
            margin: 0;
        }

        label{
            margin-left: 0;
            display: inline-block;
            margin-bottom: 0.2rem;
        }
    }

    .register__options--user{
        text-align: center;
    }
   
    .register__form .button {
        margin: 0 auto;
        margin-top: 4vh;
        height: 30px;
    }
}
