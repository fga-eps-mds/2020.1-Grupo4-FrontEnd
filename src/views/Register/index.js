import React from 'react';
import Button from '../../components/Button';
import '../../index.css'
import './style.scss';
import { useForm } from 'react-hook-form';
import { registerRequest } from '../../services/usersService';

function Register() {

    const { handleSubmit, register, watch, errors } = useForm();
    const onSubmit = registerRequest

    return (
        <>
            <div id="register">
                <div className="register__form">
                    <h1>Cadastro</h1>
                    <label>
                        <form onSubmit={handleSubmit(onSubmit)}>
                            <div className="register__inputs">
                                <div className="register__grid1">
                                    <p>nome</p><input type="text" name="name" ref={register({
                                        required: "campo obrigatório",
                                        pattern: {
                                            value: /^[A-Za-z][A-Za-z\s]*$/,
                                            message: "Nome inválido"
                                        }
                                    })} />
                                    {(errors.name && <span className="danger">{errors.name.message}</span>) || <br />}
                                    <p>email</p> <input
                                        type="text"
                                        name="email"
                                        placeholder="email@email.com"
                                        ref={register({
                                            required: "campo obrigatório",
                                            pattern: {
                                                value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
                                                message: "email inválido"
                                            }
                                        })}
                                    />
                                    {(errors.email && <span className="danger">{errors.email.message}</span>) || <br />}
                                </div>
                                <div className="register__grid2">
                                    <p>senha</p> <input
                                        type="password"
                                        name="password"
                                        ref={register({
                                            required: "campo obrigatório",
                                            minLength: {
                                                value: 6,
                                                message: "tamanho mínimo é 6"
                                            }
                                        })}
                                    />
                                    {(errors.password && <span className="danger">{errors.password.message}</span>) || <br />}

                                    <p>confirmar senha</p> <input
                                        type="password"
                                        name="confirmPassword"
                                        ref={register({
                                            required: "campo obrigatório",
                                            minLength: {
                                                value: 6,
                                                message: "tamanho mínimo é 6",
                                            },
                                            validate: (value) => {
                                                const passwordMatches = value === watch('password');
                                                return passwordMatches ? passwordMatches : "As senhas nao coincidem"  
                                            }
                                        })}

                                    />
                                    {(errors.confirmPassword && <span className="danger">{errors.confirmPassword.message} </span>) || <br />}
                                </div>
                            </div>
                            <div className="register__options">
                                <div className="register__options--user">
                                    <p>tipo de cadastro</p>
                                    <label htmlFor="mentor"><input name="userType" value="mentor" type="radio" ref={register} /> mentora</label>
                                    <label htmlFor="learner"><input name="userType" value="aprendiz" type="radio" ref={register} /> aprendiz</label>
                                </div>
                                <label htmlFor="agree"><input name="agree" type="checkbox" ref={register({
                                    required: "concorde com os termos de uso",

                                })} /> Concordo que este tutorial é somente para aqueles que se identificam com o gênero feminino.</label>
                                <label htmlFor="terms"><input name="terms" type="checkbox" ref={register({
                                    required: "concorde com os termos de uso",
                                })} /> Concordo com os termos de uso.</label>
                                {errors.terms && <span className="danger">{errors.terms.message}</span>}
                            </div>
                            <Button>REGISTRAR</Button>
                        </form>
                    </label>
                </div>
            </div>
        </>
    );
}

export default Register;