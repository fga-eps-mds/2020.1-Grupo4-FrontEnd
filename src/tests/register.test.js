import { shallow } from 'enzyme';
import React from 'react';
import Register from '../views/Register';
import 'mutationobserver-shim';
import { store } from '../store'
global.MutationObserver = window.MutationObserver;

it('Expect to render Register Component', () => {
  expect(shallow(<Register store={store}></Register>).length).toEqual(1)
  expect(shallow(<Register></Register>)).toMatchSnapshot()
})

it('Expect to click button', () => {
  const wrapper = shallow(<Register store={store} />);
  wrapper.find('Button').simulate('click')
})

it('Expect to click name input', () => {
    const wrapper = shallow(<Register store={store}/>);
    const input = wrapper.find('input').at(0).simulate('change', { target: { name: 'name' } })
    expect(input.props().name).toEqual('name');
})

it('Expect to click email input', () => {
    const wrapper = shallow(<Register store={store} />);
    const input = wrapper.find('input').at(1).simulate('change', { target: { name: 'email' } })
    expect(input.props().name).toEqual('email');
})

it('Expect to click password input', () => {
    const wrapper = shallow(<Register store={store}/>);
    const input = wrapper.find('input').at(2).simulate('change', { target: { name: 'password' } })
    expect(input.props().name).toEqual('password');
})