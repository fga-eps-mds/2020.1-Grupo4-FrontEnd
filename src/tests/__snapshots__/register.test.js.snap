// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Register Component 1`] = `
<Fragment>
  <div
    id="register"
  >
    <div
      className="register__form"
    >
      <h1>
        Cadastro
      </h1>
      <label>
        <form
          onSubmit={[Function]}
        >
          <div
            className="register__inputs"
          >
            <div
              className="register__grid1"
            >
              <p>
                nome
              </p>
              <input
                name="name"
                type="text"
              />
              <br />
              <p>
                email
              </p>
               
              <input
                name="email"
                placeholder="email@email.com"
                type="text"
              />
              <br />
            </div>
            <div
              className="register__grid2"
            >
              <p>
                senha
              </p>
               
              <input
                name="password"
                type="password"
              />
              <br />
              <p>
                confirmar senha
              </p>
               
              <input
                name="confirmPassword"
                type="password"
              />
              <br />
            </div>
          </div>
          <div
            className="register__options"
          >
            <div
              className="register__options--user"
            >
              <p>
                tipo de cadastro
              </p>
              <label
                htmlFor="mentor"
              >
                <input
                  name="userType"
                  type="radio"
                  value="mentor"
                />
                 mentora
              </label>
              <label
                htmlFor="learner"
              >
                <input
                  name="userType"
                  type="radio"
                  value="aprendiz"
                />
                 aprendiz
              </label>
            </div>
            <label
              htmlFor="agree"
            >
              <input
                name="agree"
                type="checkbox"
              />
               Concordo que este tutorial é somente para aqueles que se identificam com o gênero feminino.
            </label>
            <label
              htmlFor="terms"
            >
              <input
                name="terms"
                type="checkbox"
              />
               Concordo com os termos de uso.
            </label>
          </div>
          <Button>
            REGISTRAR
          </Button>
        </form>
      </label>
    </div>
  </div>
</Fragment>
`;
